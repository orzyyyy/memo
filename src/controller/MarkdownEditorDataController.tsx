import React from 'react';
import ReactDOM from 'react-dom';
import { message } from 'antd';
import { useFetchDocumentData } from '../hooks/useFetchDocumentData';
import MarkdownEditor from '../pages/MarkdownEditor';
import { DataSource } from 'mini-xmind/lib/canvas/core';
import { getPathNameFromUrl } from '../utils';

export interface MarkdownEditorDataControllerProps {
  match: any;
}
export interface MarkdownEditorDataControllerState {
  dataSource: string;
}
export interface MarkdownEditorSaveProps {
  id: string;
  layout: DataSource;
  format?: boolean;
  category: 'mapping' | 'markdown';
}

export function showMessageAfterFetching(result: any) {
  result ? message.success('保存成功') : message.error('保存失败');
}

const MarkdownEditorDataController = () => {
  const id: string = getPathNameFromUrl();

  const handleOnSave = async () => {
    const params: MarkdownEditorSaveProps = {
      id,
      layout: data,
      category: 'markdown',
      format: true,
    };
    const response = await fetch('../document/update', {
      body: JSON.stringify(params),
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
    });
    const result = await response.json();
    showMessageAfterFetching(result);
  };

  const [data, setData] = useFetchDocumentData(id, 'markdown', handleOnSave);

  const handleOnChange = (value: string) => {
    setData(value);
  };

  return (
    <MarkdownEditor
      value={data}
      onChange={handleOnChange}
      onSave={handleOnSave}
    />
  );
};

ReactDOM.render(
  <MarkdownEditorDataController />,
  document.getElementById('root'),
);
