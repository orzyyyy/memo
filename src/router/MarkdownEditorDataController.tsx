import React, { Suspense, lazy } from 'react';
import ReactDOM from 'react-dom';
import TohoLoading from '../pages/TohoLoading';

const MarkdownEditorDataController = lazy(() =>
  import('../controller/MarkdownEditorDataController'),
);

ReactDOM.render(
  <Suspense fallback={<TohoLoading />}>
    <MarkdownEditorDataController />
  </Suspense>,
  document.getElementById('root'),
);
