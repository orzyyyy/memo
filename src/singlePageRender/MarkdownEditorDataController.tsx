import React from 'react';
import ReactDOM from 'react-dom';
import MarkdownEditorDataController from '../controller/MarkdownEditorDataController';

ReactDOM.render(
  <MarkdownEditorDataController />,
  document.getElementById('root'),
);
