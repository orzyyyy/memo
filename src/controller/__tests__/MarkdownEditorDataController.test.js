import React from 'react';
import { mount } from 'enzyme';
import MarkdownEditorDataController from '../MarkdownEditorDataController';
import fetchMock from 'fetch-mock';

describe('MarkdownEditorDataController', () => {
  const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});

  beforeEach(() => {
    fetchMock.mock('/document/update', {});
    fetchMock.mock('/markdown//.md', { test: 1 });
  });

  afterAll(() => {
    errorSpy.mockRestore();
  });

  afterEach(() => {
    fetchMock.restore();
  });

  it('handleOnSave', async () => {
    const wrapper = mount(<MarkdownEditorDataController />);
    await wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });
});
