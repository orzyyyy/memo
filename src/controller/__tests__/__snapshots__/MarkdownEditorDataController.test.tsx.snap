// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownEditorDataController render correctly 1`] = `
<MarkdownEditorDataController>
  <MarkdownEditor
    onChange={[Function]}
    onSave={[Function]}
  >
    <div
      style={
        Object {
          "display": "grid",
          "fontSize": 16,
          "gridTemplateColumns": "50% 50%",
          "padding": 24,
        }
      }
    >
      <textarea
        className="markdown-body"
        onChange={[Function]}
        style={
          Object {
            "marginRight": 24,
          }
        }
      />
      <div
        className="content markdown-body"
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
      />
    </div>
  </MarkdownEditor>
</MarkdownEditorDataController>
`;
